### DISCUSSION TITLE
Override wildcard DNS rewrite

### DISCUSSION THREAD

**Discussion by agneevX** [AdguardTeam/AdGuardHome#3316] [Q&A] [answered]
I have a DNS rewrite added (Filters → DNS rewrites) as `*.server → 10.0.0.11`.

I want to create a DNS rewrite that returns `0.0.0.0` for one particular host:

```regex
# Custom filtering rules

||null-host.server^$important,dnsrewrite=0.0.0.0

# Returns 10.0.0.11
```

If I delete the rewrite and add a manual rule rule for it:

```regex
# Custom filtering rules

||*.server^$dnsrewrite=10.0.0.11
||null-host.server^$important,dnsrewrite=0.0.0.0

# Returns 0.0.0.0 and 10.0.0.11
```

Is there a way I can have it reply `0.0.0.0` only?

**Comment by ameshkov** [comment, marked as answer]
If you want to target just a single hostname and not it's subdomains, you'd better do something like that:
`|null-host.server^$dnsrewrite=0.0.0.0`

`|` - points to a beginning of a hostname, it does not match subdomains like `||` does.
For the second rule, you should use `denyallow`: `||*.server^$dnsrewrite=10.0.0.11,denyallow=null-host.server `

Here's the pair of rules that does what you need:

```
||*.server^$dnsrewrite=10.0.0.11,denyallow=null-host.server
||null-host.server^$dnsrewrite=0.0.0.0
```
